### ISSUE TITLE
Support for collecting active network connections

### ISSUE THREAD

**Issue by puffyCid** [puffyCid/artemis#108]
**What new feature do you think would be cool to add to artemis?**
Artemis currently does not support collecting active network connections. It would be really cool if an user could collect network connections and associated process information. (similar to netstat)

**Describe the solution you'd like**
Collect active network connections

**Additional context**
Since active network connections only exist in memory, we will need to use OS APIs to collect them.
The ideal solution is to find a Rust crate that supports collecting network connections on Windows, macOS, and Linux

**Comment by puffyCid** [comment]
https://github.com/GyulyVGC/listeners could be an interesting library that could provide this functionality.
however, right now it only supports processes listening for TCP connections

**Comment by scott91e1** [comment]
Would it be possible to have a collector that runs an external command and have a JS that parses the well formatted stdout into a format digestible by artemis?

**Comment by puffyCid** [comment]
Yes running external command is [possible](https://puffycid.github.io/artemis-api/docs/API/Helper/system#executecommandcommand-args---commandresult--error) via the JS API.

However, shelling out directly from Rust code is not allowed. [Contributing doc](https://github.com/puffycid/artemis/blob/main/CONTRIBUTING.md)

For network connections, OS APIs will have to be used. Since this data only exists in memory.
